pycparser/c_generator.py

```diff
@@ -289,23 +278,55 @@ def visit_NamedInitializer(self, n):
     def visit_FuncDecl(self, n):
         return self._generate_type(n)
 
-    def _generate_struct_union(self, n, name):
-        """ Generates code for structs and unions. name should be either
-            'struct' or union.
+    def _generate_struct_union_enum(self, n, name):
+        """ Generates code for structs, unions, and enums. name should be
+            'struct', 'union', or 'enum'.
         """
+        if name in ('struct', 'union'):
+            members = n.decls
+            body_function = self._generate_struct_union_body
+        elif name in ('enum',):
+            if n.values is None:
+                members = ()
```

Comment:
`members = () if n.values is None else n.values.enumerators`

Reply:
Pending above possibilities I'll change this.